darts/tests/explainability/test_shap_explainer.py

```diff
@@ -139,17 +136,87 @@ class TestShapExplainer:
     )
 
     def test_creation(self):
-        model_cls = LightGBMModel if lgbm_available else XGBModel
-        # Model should be fitted first
-        m = model_cls(
-            lags=4,
-            lags_past_covariates=[-1, -2, -3],
-            lags_future_covariates=[0],
-            output_chunk_length=4,
-            add_encoders=self.add_encoders,
-        )
-        with pytest.raises(ValueError):
-            ShapExplainer(m, self.target_ts, self.past_cov_ts, self.fut_cov_ts)
+        # Tests only for gradient boosting models
+        if GBM_AVAILABLE:
```

Comment:
Having everything here in an `if` scope seems a bit like we should rather create a dedicated test for the GBMs. Could you move that logic into a new test and parametrize it so we can avoid code duplication? Would be nice to test it for all xgb, lgbm and catboost.